### PR TITLE
test(bigtable): run against live endpoint for continuous tests

### LINKED ISSUE #4158: bigtable: run tests against live bigtable for continuous test runs
today the bigtable tests are running under emulation for the continuous runs. As some tests require live bigtable, we should use the live instance for continuous.

### PR DESCRIPTION
**Pull request by crwilcox**
Fixes #4158 (target live service for continuous tests) and #4157  (cleanup)

### CODE REVIEW

**Comment by crwilcox** [comment]
Manual continuous run is green 🥳

**Review by kolea2 — APPROVED** [review]
this is awesome!

File: bigtable/integration_test.go [hunk 1]
```diff
@@ -96,7 +100,48 @@ func TestMain(m *testing.M) {
...
 	}
-	os.Exit(m.Run())
+	exit := m.Run()
+	if err := cleanup(c); err != nil {
+		log.Printf("Post-test cleanup failed: %v", err)
+	}
+	os.Exit(exit)
+}
+
+func cleanup(c IntegrationTestConfig) error {
```
Comment (tritone) [inline bigtable/integration_test.go:110]: Nice, thanks for adding this.

File: bigtable/integration_test.go [hunk 2]
```diff
@@ -80,14 +83,15 @@ var instanceToCreate string
...
-		instanceToCreate = fmt.Sprintf("create-%d", time.Now().Unix())
+		instanceToCreate = fmt.Sprintf("bt-it-%d", time.Now().Unix())
 	}
 }
 
 func TestMain(m *testing.M) {
 	flag.Parse()
 
-	c := integrationConfig
+	env, _ := NewIntegrationEnv()
```
Comment (tritone) [inline bigtable/integration_test.go:93]: You should receive the error from `NewIntegrationEnv` and fail on it if it's not nil.

File: bigtable/integration_test.go [hunk 3]
```diff
@@ -96,7 +100,48 @@ func TestMain(m *testing.M) {
...
+			timestamp := info.Name[len(instancePrefix):]
+			t, err := strconv.ParseInt(timestamp, 10, 64)
+			if err != nil {
+				return err
+			}
+			uT := time.Unix(t, 0)
+			sevenDays := time.Hour * 24 * 7
+			if time.Now().After(uT.Add(sevenDays)) {
+				// delete an instance older than 7 days
+				iac.DeleteInstance(ctx, info.Name)
```
Comment (tritone) [inline bigtable/integration_test.go:139]: I think even a shorter timeframe than 7 days is probably good given how expensive bigtable instances are, no? Also, is it just instances that need to be deleted, or any other resources as well?

**Review by tritone — COMMENTED** [review]
Good improvements, thanks!

File: bigtable/integration_test.go (hunk 3 above)
Comment (crwilcox) [reply bigtable/integration_test.go:139]: There are instances and clusters, but they are coupled, so 99% sure I don't need to also call delete cluster. @kolea2 can confirm likely.

Comment (kolea2) [reply bigtable/integration_test.go:139]: yep! https://cloud.google.com/bigtable/docs/deleting-instance

File: bigtable/integration_test.go (hunk 2 above)
Comment (crwilcox) [reply bigtable/integration_test.go:93]: I have altered this to panic if this happens with an error.

File: bigtable/integration_test.go [hunk 4]
```diff
@@ -96,7 +103,46 @@ func TestMain(m *testing.M) {
...
+	if err := cleanup(c); err != nil {
+		log.Printf("Post-test cleanup failed: %v", err)
+	}
+	os.Exit(exit)
+}
+
+func cleanup(c IntegrationTestConfig) error {
+	// Cleanup resources marked with bt-it- after a time delay
+	timeToWait := time.Hour * 48 // two days
+	instancePrefix := "bt-it-"
```
Comment (codyoss) [inline bigtable/integration_test.go:116]: nit: could this be a const so the cleanup and create can stay in sync should one change over time.

**Pull request merged**